outlook/src/taskpane/components/Login/Login.tsx

```diff
@@ -45,76 +68,157 @@ class Login extends React.Component<{}, LoginState> {
+     * code if the user accepted the permission.
+     */
+    _openOdooLoginDialog = async () => {
         const options = {
-        height: 65,
-        width: 30,
-        promptBeforeOpen: true,
+            height: 65,
+            width: 30,
+            promptBeforeOpen: true,
         };
 
         const redirectToAddin = encodeURIComponent(api.addInBaseURL + '/taskpane.html');
-        const redirectToAuthPage = encodeURIComponent(api.authCodePage + '?scope=' + api.outlookScope + '&friendlyname=' + api.outlookFriendlyName + '&info=' + '&redirect=' + redirectToAddin);
+        const redirectToAuthPage = encodeURIComponent(api.authCodePage + '?scope=' + api.outlookScope + '&friendlyname=' + api.outlookFriendlyName + '&redirect=' + redirectToAddin);
         const loginURL = api.baseURL + api.loginPage + '?redirect=' + redirectToAuthPage;
+        const url = api.addInBaseURL + '/dialog.html?dialogredir=' + loginURL;
 
-        Office.context.ui.displayDialogAsync(api.addInBaseURL + '/dialog.html?dialogredir=' + loginURL, options , (asyncResult) => {
-            let dialog = asyncResult.value;
-            dialog.addEventHandler(Office.EventType.DialogMessageReceived, (_arg) => {
-                dialog.close();
-                let success = new URL(JSON.parse(_arg['message']).value).searchParams.get("success");
-                if (success === "1")
-                {
-                    let code = new URL(JSON.parse(_arg['message']).value).searchParams.get("auth_code");
-                    sendHttpRequest(HttpVerb.POST, api.baseURL + api.getAccessToken, ContentType.Json, null, {"auth_code": code}, true)
-                        .promise.then((response) => {
-                        const parsed = JSON.parse(response);
-                        this.context.connect(parsed.result.access_token);
-                        this.context.navigation.goToMain();
-                    });
-                }
+        return new Promise((resolve, reject) => {
+            Office.context.ui.displayDialogAsync(url, options , (asyncResult) => {
+                const dialog = asyncResult.value;
+                dialog.addEventHandler(Office.EventType.DialogMessageReceived, (_arg) => {
+                    dialog.close();
+                    const searchParams = new URL(JSON.parse(_arg['message']).value).searchParams;
+                    const success = searchParams.get("success");
+                    if (success === "1") {
+                        const authCode = searchParams.get("auth_code");
+                        resolve(authCode);
+                    } else {
+                        reject();
+                    }
+                });
             });
-        })
-    };
+        });
+    }
+
+    /**
+     * Make an HTTP request to the Odoo database to exchange the authentication code
+     * for a long term access token.
+     *
+     * Return the access token or raise an error is something went wrong.
```

Comment:
if*

Reply:
oops, one day I'll find a grammar checker for sublime text...